src/libutil/unix/environment-variables.cc

```diff
@@ -1,21 +1,12 @@
-#include "util.hh"
-#include "environment-variables.hh"
+#include <stdlib.h>
 
-extern char * * environ __attribute__((weak));
+#include "environment-variables.hh"
 
 namespace nix {
 
-void clearEnv()
-{
-    for (auto & name : getEnv())
-        unsetenv(name.first.c_str());
-}
-
-void replaceEnv(const std::map<std::string, std::string> & newEnv)
+int setenv(const char * name, const char * value)
```

Comment:
```suggestion
int setEnv(const char * name, const char * value)
```

That way it's clear that it's not the POSIX `setenv`, and it's more coherent with our naming scheme

Reply:
(ideally it should be 

```suggestion
void setEnv(const std::string & name, const std::string & value)
```

and throw in case of errors, but let's indeed save that for a follow-up as you mentioned
)